client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -273,25 +273,21 @@ private synchronized ErrorCode startTunnel(
     try {
       client = new shadowsocks.Client(configCopy);
     } catch (Exception e) {
-      LOG.log(Level.WARNING, "Invalid configuration", e);
+      LOG.log(Level.WARNING, "shadowsocks.Client initialization failed", e);
       tearDownActiveTunnel();
-      return ErrorCode.ILLEGAL_SERVER_CONFIGURATION;
+      return e.getMessage();
     }
 
-    ErrorCode errorCode = ErrorCode.NO_ERROR;
+    boolean udpSupported = false;
     if (!isAutoStart) {
       try {
         // Do not perform connectivity checks when connecting on startup. We should avoid failing
         // the connection due to a network error, as network may not be ready.
-        errorCode = checkServerConnectivity(client);
-        if (!(errorCode == ErrorCode.NO_ERROR
-                || errorCode == ErrorCode.UDP_RELAY_NOT_ENABLED)) {
-          tearDownActiveTunnel();
-          return errorCode;
-        }
+        udpSupported = checkUDPConnectivity(client);
```

Comment:
No I am not. The teardown behavior is covered in the `catch` clause, because `checkUDPConnectivity` will throw an `Exception` if TCP is not supported.